bc/subscription/utils/courtlistener.py

```diff
@@ -0,0 +1,139 @@
+import logging
+
+import courts_db
+import requests
+from django.conf import settings
+
+from .exceptions import MultiDefendantCaseError
+
+logger = logging.getLogger(__name__)
+
+CL_API = {
+    "docket": "https://www.courtlistener.com/api/rest/v3/dockets/",
+    "docket-alerts": "https://www.courtlistener.com/api/rest/v3/docket-alerts/",
+}
+
+
+def lookup_court(court: str):
+    """
+    Lookup a court name or citation string using courts-db.
+    Returns court ID (e.g., "cand" for "N.D. Cal.")
+    Returns None if it can't find anything.
+    """
+    results = courts_db.find_court(court)
+    if len(results) == 1:
+        return results[0]
+    elif len(results) == 0:
+        print(f"No results for court '{court}'")
+        return None
+    else:
+        print(f"Could not resolve court '{court}'")
+        return None
+
+
+def auth_header() -> dict:
+    token = settings.COURTLISTENER_API_KEY
+    header_dict = {"Authorization": f"Token {token}"}
+    return header_dict
+
+
+def lookup_docket_by_cl_id(cl_id: int):
+    """
+    Performs a GET query on /api/rest/v3/dockets/
+    to get a Docket using the CourtListener ID
+    """
+    url = f"{CL_API['docket']}{cl_id}/"
+    response = requests.get(url, headers=auth_header(), timeout=5)
+    if response.status_code == 200:
+        return response.json()
+    else:
+        return None
```

Comment:
I feel like this should do a `raise_for_status()` or something instead of just returning None when it's not a 200. Like, somebody is going to be pretty annoyed when they realize that the API failed but didn't say why or crash loudly.

Reply:
You're right. I'll use `raise_for_status()` and let the app crash loudly